### PR TITLE
Migrate various tests to JUnit

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/operator/AnnotationEngineAssertions.java [hunk 1]
```diff
@@ -18,21 +18,25 @@
 
 import static org.testng.Assert.assertEquals;
 
-abstract class TestAnnotationEngine
+class AnnotationEngineAssertions
 {
-    void assertImplementationCount(ParametricScalar scalar, int exact, int specialized, int generic)
+    private AnnotationEngineAssertions()
+    {
+    }
```
Comment (dain) [inline core/trino-main/src/test/java/io/trino/operator/AnnotationEngineAssertions.java:25]: one line

**Review by dain — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java [hunk 2]
```diff
@@ -326,11 +326,11 @@ private static Object convertToRowValue(Type type, Object value)
         }
         if (type.getBaseName().equals("Geometry")) {
             //noinspection RedundantCast
-            return (byte[]) value;
+            return (String) value;
```
Comment (findepi) [inline testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java:329]: How does a change of a testing framework affect the data representation in the protocol?

File: plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestSpatialJoinPlanning.java [hunk 3]
```diff
@@ -183,12 +196,12 @@ public void testSpatialJoinWithin()
...
                                 anyTree(
                                         unnest(
-                                                project(ImmutableMap.of("partitions", expression(format("spatial_partitions(%s, st_point)", kdbTreeLiteral))),
+                                                project(ImmutableMap.of("partitions_a", expression(format("spatial_partitions(%s, st_point)", kdbTreeLiteral))),
                                                         project(ImmutableMap.of("st_point", expression("ST_Point(lng, lat)")),
                                                                 tableScan("points", ImmutableMap.of("lng", "lng", "lat", "lat", "name_a", "name")))))),
                                 anyTree(
                                         unnest(
-                                                project(ImmutableMap.of("partitions", expression(format("spatial_partitions(%s, st_geometryfromtext)", kdbTreeLiteral))),
+                                                project(ImmutableMap.of("partitions_b", expression(format("spatial_partitions(%s, st_geometryfromtext)", kdbTreeLiteral))),
```
Comment (findepi) [inline plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestSpatialJoinPlanning.java:204]: How does a change of a testing framework affect how the plan matchers work?

or is there something else contributors should now be aware of?

File: testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java (hunk 2 above)
Comment (martint) [reply testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java:329]: It does not. This should've been a separate commit in this PR, but I must have squashed it with the other one by mistake. The problem was that the test was broken but it wasn't running in the CI.

File: plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestSpatialJoinPlanning.java (hunk 3 above)
Comment (martint) [reply plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestSpatialJoinPlanning.java:204]: Same here.

File: testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java (hunk 2 above)
Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java:329]: I confirm at 427  tag the geo tests were not passing and become passing with this change.

Do you know how long was it broken? 
it seems we had `return (byte[]) value;` "for ever" here and i think geo tests used to be passing with it in older releases.

Comment (martint) [reply testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java:329]: I'm not sure. I checked a few releases back and they were not passing either, but didn't keep digging to see when that happened.

Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java:329]: In Trino 400, `TestGeoSpatialQueries` works.
In 405 it was probably already not executing on CI, but stil works.
In 410 the test fails with class cast exception here.

File: core/trino-spi/src/test/java/io/trino/spi/security/TestIdentity.java [hunk 4]
```diff
@@ -48,26 +47,26 @@ public void testEquals()
...
-                {
-                        {Identity.from(TEST_IDENTITY).withPrincipal(new BasicPrincipal("other principal")).build()},
-                        {Identity.from(TEST_IDENTITY).withGroups(ImmutableSet.of("group2", "group3")).build()},
-                        {Identity.from(TEST_IDENTITY).withEnabledRoles(ImmutableSet.of("role2", "role3")).build()},
-                        {Identity.from(TEST_IDENTITY).withConnectorRoles(ImmutableMap.of(
-                                "connector2", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("connector2role")),
-                                "connector3", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("connector3role"))))
-                                .build()},
-                };
+        assertThat(Identity.from(TEST_IDENTITY).withPrincipal(new BasicPrincipal("other principal")).build())
```
Comment (breglend) [inline core/trino-spi/src/test/java/io/trino/spi/security/TestIdentity.java:56]: Is this not identical to the assertion on line 53? Is this duplication in error?

**Pull request merged**